lib/improver/cli/nowcast_extrapolate.py

```diff
@@ -173,17 +76,18 @@ def process(input_cube, u_cube=None, v_cube=None, speed_cube=None,
         ValueError:
             can either use speed_cube and direction_cube or u_cube and v_cube.
     """
-    if (speed_cube or direction_cube) and (u_cube or v_cube):
-        raise ValueError('Cannot mix advection component velocities with speed'
-                         ' and direction')
-    if not (speed_cube and direction_cube) and not (u_cube and v_cube):
-        raise ValueError('Either speed and direction or u and v cubes '
-                         'are needed.')
+    from iris import Constraint
 
-    if speed_cube and direction_cube:
-        u_cube, v_cube = ResolveWindComponents().process(
-            speed_cube, direction_cube)
+    from improver.nowcasting.forecasting import CreateExtrapolationForecast
+    from improver.utilities.cube_manipulation import merge_cubes
 
+    u_cube = advection_cubes.extract(
+        Constraint("precipitation_advection_x_velocity"), True)
```

Comment:
I think it might be useful to have the expected cube names for the advection velocities in the docstring.